firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/chunks.txt

```diff
@@ -0,0 +1,262 @@
+  // output buffer is empty
+
+    WHEN("257th non-delimiter byte is passed as input")
+      THEN("The input method reports invalid_length status")
+
+    WHEN("More than 256 non-delimiter bytes are passed as input")
+      THEN("The input method reports invalid_length status on 257th byte and for all the subsequent bytes")
+    
+    WHEN("More than 256 non-delimiter bytes are passed as input, after which a delimiter is passed and output is called")
+      THEN("The input method reports invalid_length status for all the 256 non-delimiter bytes")
+      THEN("The input method reports output ready status on the delimeter byte")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+    WHEN("a non-delimiter byte is passed as input and output is called")
+      THEN("The input method reports invalid_length status")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("The input method reports output_ready status")
+      THEN("the output method reports ok status")
+      THEN("The output buffer is as expected")
+
+Scenario: ChunkSplitter constructed with with default value for delimeter and include_delimeter boolean as true correctly handles unacceptably long input chunks
+
+  GIVEN("A uint8_t chunk splitter with a completely full internal buffer of 256 bytes")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("257th non-delimiter byte is passed as input")
+      THEN("The input method reports invalid_length status")
+
+    WHEN("More than 256 non-delimiter bytes are passed as input")
+      THEN("The input method reports invalid_length status on 257th byte and for all the subsequent bytes")
+
+    WHEN("More than 256 non-delimiter bytes are passed as input, after which a delimiter is passed and output is called")
+      THEN("The input method reports invalid_length status for all the 256 non-delimiter bytes")
+      THEN("The input method reports output ready status on the delimeter byte")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+    WHEN("a non-delimiter byte is passed as input and output is called")
+      THEN("The input method reports invalid_length status")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+    WHEN("A delimeter byte is passed as input and output is called")
+      THEN("The input method reports invalid_length status")
+      THEN("the output method reports invalid_length status")
+      THEN("The output buffer is as expected")
+
+Scenario: After output is consumed, ChunkSplitter's behavior resets to that of a ChunkSplitter with an empty internal buffer
+
+  GIVEN("A uint8_t ChunkSplitter completely filled with input data, and no delimiters are consumed with an output call")
```

Comment:
With the changes we've made to the input method, it's no longer possible to fill the internal buffer with inputs including multiple delimiter bytes inside the internal buffer. I think we can rephrase this like "GIVEN a uint8_t ChunkSplitter of capacity 256 bytes which has received 256 bytes as input, including multiple delimiter bytes".

However, I think this GIVEN should be moved into a different scenario (the scenario testing the input_overwritten feature), and other tests should be defined for this scenario, e.g. starting from an empty ChunkSplitter, filling its buffer, consuming with output, and then being able to completely fill its buffer again. We could even do this in a `for` loop for _n_ iterations.